examples/PerfectPowers.jl

```diff
@@ -302,13 +316,12 @@ function ispower_bernstein(a::fmpz)
   c = [x for x = c if !isone(x)]
   #not sure still necessary
   @time c = Hecke.coprime_base(c) 
-  @show length(c), length(Set(c))
   k = [valuation(a, p) for p = c]
   @show no_p
   return gcd(k)*f
```

Comment:
In the  base cases, this function seems to return a tuple `(n,x)` such that the `x^n == a`. But here a single integer (?) is returned? How does that fit together?